packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -418,6 +418,20 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange>
 		const enrichedCommit = this.resubmitMachine.peekNextCommit();
 		this.submitCommit(enrichedCommit, localOpMetadata, true);
 	}
+	public rollback(content: JsonCompatibleReadOnly, localOpMetadata: unknown): void {
+		// Empty context object is passed in, as our decode function is schema-agnostic.
+		const {
+			commit: { revision },
+		} = this.messageCodec.decode(this.serializer.decode(content), {
+			idCompressor: this.idCompressor,
+		});
+		const [commit] = this.editManager.findLocalCommit(revision);
+		const { parent } = commit;
+		assert(parent !== undefined, "must have parent");
+		const [precedingCommit] = this.editManager.findLocalCommit(parent.revision);
+		this.editManager.localBranch.removeAfter(precedingCommit);
```

Comment:
this seems to work, but i'm not sure it is correct. Rollback always starts at the end of the pending queue and works backwards, is removeAfter the right thing to do? will repeated calls on each pending item in the queue have any negative effects?

Reply:
That approach makes sense to me. I think the downside of repeated calls (compared to just rolling back all in-flight commits) is purely the perf overhead. How bad the perf overhead turns out to be largely depends on the app. Seems good enough for a first cut though.